arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorHolder.java

```diff
@@ -59,12 +59,17 @@ public VectorHolder(
 
   // Only used for returning dummy holder
   private VectorHolder() {
+    this(null);
```

Comment:
I will need to think more about untyped null holders. I am not sure it is a good idea to have them.

Reply:
cc @rdblue @RussellSpitzer @nastra @flyrain

Reply:
While it is arguable, the current solution works as null checks are performed prior to accessing values. Supporting typed null vectors would be a substantial change in our Arrow codebase and I am not convinced it would be worth it. Keeping as-is for now.

Reply:
Exposing fields directly would cause a checkstyle violation. I feel the current approach is simple enough.